GetStream/stream-chat-swift
# [CIS-1734] Paginated channels are left without messages when sync is executed

### 🔗 Issue Links

https://stream-io.atlassian.net/browse/CIS-1734

### 🎯 Goal

When in a paginated channel list, if the sync was executed (eg. backgrounding the app and opening it), the channels were left in the list but showing "No messages".

### 🛠 Implementation

The issue described above was caused because we were only cleaning those channels, instead of deleting them.
Not we are properly deleting them.
While fixing this issue, other issues surfaced, which are tackled in #1984 

### 🧪 Manual Testing Notes

Pre-requisite: Make sure chat is disconnected when going to background

1. Open the DemoApp
2. Select a user
3. Scroll to paginate channels list
4. Background the app
5. Reopen the app

Expected result:
- Channels out of the first page should be deleted.

### ☑️ Contributor Checklist

### 🎁 Meme

_Provide a funny gif or image that relates to your work on this pull request. (Optional)_
